### ISSUE TITLE
switch to a new release model

### ISSUE THREAD

**Issue by kazuho** [h2o/h2o#3230]
As people have noticed, we have long failed to tag new versions, even though we are rapidly adding features.

We plan to switch to a new release model where every commit to master is considered ready for production.

But simply doing that is going to stagnate our efforts to extend h2o, as there is high bar for calling something production-ready.

Therefore, as we switch to the new model, each feature will be designated either production-ready or not.

New features will be initially added as "experimental" or "under-development," then if it becomes stable and worthwhile to maintain, designated as "production-ready."

**Comment by utrenkner** [comment]
@kazuho Thank you for addressing the release model. While clear releases would be nice for us  "consumers", I fully understand your hesitation to call something production-ready. Therefore, the new model is a good compromise I can work with.

**Comment by dch** [comment]
Thanks for making changes to the model, it's much more clear for people going forwards. Should we consider the old (tagged) releases as now deprecated, for example wrt security fixes?

**Comment by trendymail** [comment]
@kazuho; this is great news! :)

For several years, I just "picked" (after reading some commits) a master release without knowing if it was "production ready" or not.

I fully understand the major overhead to maintain this new model (experimental/development/production).

But for us, people not able to understand the source code, this is really a major shift.

This way, it becomes possible to download:
- production releases => when someone needs to upgrade in a hurry
- experimental/development releases => to use cutting-edge functionalities

In my case, I would like to use and test latest changes about HTTP3.

No worries if h2o could crash sometimes and/or randomly because it is restarted immediately.

So, If something goes wrong, I will be able to submit issues if it is relevant and useful.

Have a great day!

**Comment by kazuho** [comment]
@dch 
> Should we consider the old (tagged) releases as now deprecated, for example wrt security fixes?

Yes. There is no plan to provide security patches for the tagged releases.
